apps/extension/src/components/login-form.tsx

```diff
@@ -0,0 +1,56 @@
+import { Button, InputProps } from 'ui/components';
+import { PasswordInput } from './password-input';
+import { useStore } from '../state';
+import { passwordSelector } from '../state/password';
+import { useState } from 'react';
+import { usePageNav } from '../utils/navigate';
+import { PagePath } from '../routes/page/paths';
+
+export const LoginForm = () => {
+  const navigate = usePageNav();
+
+  const { setPassword, isPassword } = useStore(passwordSelector);
+  const [input, setInputValue] = useState('');
+  const [enteredIncorrect, setEnteredIncorrect] = useState(false);
+
+  const handleUnlock = () => {
+    void (async function () {
+      if (await isPassword(input)) {
+        setPassword(input); // saves to session state
+        navigate(PagePath.INDEX);
+      } else {
+        setEnteredIncorrect(true);
+      }
+    })();
+  };
+
+  const handleChangePassword: InputProps['onChange'] = e => {
+    setInputValue(e.target.value);
+    setEnteredIncorrect(false);
+  };
+
+  return (
+    <form onSubmit={handleUnlock} className='grid gap-4'>
```

Comment:
Your `handleUnlock` should take an event and preventdefault() right? Or does this not matter?

Reply:
I thought so, so you need to add it to `handleUnlock()`

Reply:
handleUnlock used as button handler too

Reply:
Oh, in that case, you should change the button to something like:

```
<button type="submit">Submit</button>
```

Right? And then `handleUnlock()` would only be in one place. And then you can do `e.preventDefault()` in `handleUnlock()`.

Reply:
i see